code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,826 @@
+						else
+							L.attack_animal(src)
+					else
+						if (!ckey && attackcycles >= 3) // if we've an AI who has gone through 3 infection attempts on a single target, just give up trying to infect it, and kill it instead.
+							attackstep = 5
+							L.attack_animal(src)
+							return
+						if (!IsInfected(L))
+							visible_message("<span class='notice'> \icon[src] [src] takes a moment to recover. </span>")
+							return
+						var/vdistance = 99
+						for(var/obj/machinery/atmospherics/unary/vent_pump/v in view(10,src))
+							if(!v.welded)
+								if (get_dist(src,v) < vdistance)
+									entry_vent = v
+									vdistance = get_dist(src,v)
+						var/list/numtargets = ListTargets()
+						if (numtargets.len > 0)
+							LoseTarget()
+							walk_away(src,L,2,1)
+						else if(entry_vent)
+							visible_message("<span class='notice'> \icon[src] [src] lets go of [target], and tries to flee! </span>")
+							path_to_vent = 1
+							var/temp_ai_type = ai_type
+							ai_type = 1
+							LoseTarget()
+							walk_away(src,L,2,1)
+							spawn(100) // 10 seconds
+								ai_type = temp_ai_type
+						else
+							LoseTarget()
+				else if (attackstep == 5)
+					L.attack_animal(src)
+				else
+					attackstep = 0
+			else
+				L.attack_animal(src)
+		else
+			L.attack_animal(src)
+	else
+		target.attack_animal(src)
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/adjustBruteLoss(var/damage)
+	..(damage)
+	Retaliate()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/adjustFireLoss(var/damage)
+	..(damage)
+	Retaliate()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/bullet_act(var/obj/item/projectile/Proj)
+	if (istype(Proj, /obj/item/projectile/energy/declone/declone_spider))
+		if (!degenerate)
+			if (spider_tier < 2)
+				if (ckey)
+					degenerate = 1
+				else
+					gib()
+			else
+				visible_message("<span class='danger'> \icon[src] [src] resists the bioweapon! </span>")
+	else if (istype(Proj, /obj/item/projectile/energy/declone))
+		if (!degenerate && prob(20))
+			visible_message("<span class='danger'> \icon[src] [src] looks staggered by the bioweapon! </span>")
+			if (spider_tier < 3)
+				degenerate = 1
+
+	..()
+
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: PROC OVERRIDES ---------------------------
+// --------------------------------------------------------------------------------
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/examine(mob/user)
+	..()
+	var/msg = ""
+	if (src.stat == DEAD)
+		msg += "<span class='deadsay'>It appears to be dead.</span>\n"
+	else
+		if (src.key)
+			msg += "<BR><span class='warning'>Its eyes regard you with a curious intelligence.</span>"
+		if (src.ai_type == 0)
+			msg += "<BR><span class='warning'>It appears aggressive.</span>"
+		else if (src.ai_type == 0)
+			msg += "<BR><span class='notice'>It appears defensive.</span>"
+		else if (src.ai_type == 2)
```

Comment:
this else conditonal is a waste of 5 characters, ai_type cannot `==` more than one value